### ISSUE TITLE
[Bug]: Setting the threshold using ManualThreshold seems to have no effect on the output of predict

### ISSUE THREAD

**Issue by JinYuannn** [open-edge-platform/anomalib#2229]
### Describe the bug

Setting the threshold using ManualThreshold seems to have no effect on the output of predict.
I set the `default_value` in `ManualThreshold` to 0.99 and 0.0001, but the images saved by predict are the same.

### Dataset

N/A

### Model

Reverse Distillation

### Steps to reproduce the behavior

```python
model = ReverseDistillation()

threshold = ManualThreshold(default_value=0.1)

engine = Engine(
        max_epochs=50, 
        limit_train_batches=100, 
        threshold=threshold, 
        check_val_every_n_epoch=2,
        logger=True
    )

engine.predict(
        datamodule=folder_datamodule,
        model=model,
        ckpt_path="my ckpt path" )
```

### OS information

OS information:
- OS: Windows 11
- Python version: 3.10.11
- Anomalib version: 1.2.0.dev0
- PyTorch version: 2.3.1+cu121
- CUDA/cuDNN version: 121
- GPU models and configuration: RTX 4090

### Expected behavior

different threshold, different predicted mask

### Pip/GitHub

GitHub

### Configuration YAML

```yaml
None
```

### Logs

```shell
None
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by alexriedel1** [comment]
What do your predictions look like and what do you expect? Is the code above your full code, because the model training (and validation) phase is missing.

**Comment by JinYuannn** [comment]
1.  Here is my training code:

`model = ReverseDistillation()
    
engine = Engine(
    max_epochs=200, 
    check_val_every_n_epoch=5,
    logger=True
)

# Train the model
 engine.fit(datamodule=folder_datamodule, model=model)`

2. When I finish training the model and call the model using the predict  method, no matter what value I change the default_value in ManualThreshold to, the 'Predicted Heat Map' and 'Predicted Mask' drawn by predict method are exactly the same. I believe that modifying the threshold should output different 'Predicted Mask'. For example, if I set the threshold very low, there should be many defects.

**Comment by JinYuannn** [comment]
@alexriedel1 Looking forward to your reply, thanks.

**Comment by JustRookie1** [comment]
'm also having this issue, I want to change the label of the prediction image by modifying the threshold, but ManualThreshold doesn't work.
